crates/ty_python_semantic/src/types.rs

```diff
@@ -1111,15 +1115,17 @@ impl<'db> Type<'db> {
         expected_class: ClassLiteral<'_>,
     ) -> Option<Specialization<'db>> {
         self.specialization_of_optional(db, Some(expected_class))
+            .map(|(_, specialization)| specialization)
     }
 
     fn specialization_of_optional(
```

Comment:
The name of this method vs `specialization_of` no longer makes sense, now that they differ not only in optionality of `expected_class`, but also in the return type. Maybe:
```suggestion
    fn class_and_specialization_of_optional(
```